### DISCUSSION TITLE
The weight attribute is ignored by default by centrality functions

### DISCUSSION THREAD

**Discussion by agila5** [luukvdmeer/sfnetworks#135] [Show and tell]
Hi everyone! Maybe this is quite obvious for you, but I just found out that the centrality functions defined in `tidygraph` do not automatically infer the `weight` argument defined in the edges table. 

``` r
# packages
library(sf)
#> Linking to GEOS 3.8.0, GDAL 3.0.4, PROJ 6.3.1
library(tidygraph)
#> 
#> Attaching package: 'tidygraph'
#> The following object is masked from 'package:stats':
#> 
#>     filter
library(sfnetworks)
library(tmap)

# create sfn object
sfn_roxel = as_sfnetwork(roxel, directed = FALSE)

# edge centrality measure
sfn_roxel <- sfn_roxel %E>% 
  mutate(
    weight = edge_length(),
    # actual code
    v1 = centrality_edge_betweenness(), 
    # intended code
    v2 = centrality_edge_betweenness(weights = weight, directed = FALSE)
  )

t1 <- tm_shape(sfn_roxel %E>% st_as_sf()) +  
  tm_lines(col = "v1", palette = "viridis", lwd = 2, n = 6)

t2 <- tm_shape(sfn_roxel %E>% st_as_sf()) +  
  tm_lines(col = "v2", palette = "viridis", lwd = 2, n = 6)

tmap_arrange(t1, t2)
```

![](https://i.imgur.com/xSFoYWh.png)

<sup>Created on 2021-03-10 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

I know that this is more or less the same discussion as in https://github.com/luukvdmeer/sfnetworks/discussions/67#discussioncomment-229229 (and probably the default behaviour in _tidygraph_), but if you also easily forget about this behaviour, maybe we could add a note in one of the vignettes (otherwise this is just a discussion that will help me remember this fact 😅).

**Comment by luukvdmeer** [comment]
This is indeed quite confusing.. For now I included this line in the routing vignette after the section where we talk about edge weights:

https://github.com/luukvdmeer/sfnetworks/blob/6fcd848dcc8b8e6075bb2a5dcd4749fa2dfb442b/vignettes/routing.Rmd#L66

Not sure if there are other parts of the docs where we should mention this, or we should even rethink our approach
